hack/upgrade-test-harness/conf.yaml

```diff
@@ -36,5 +36,5 @@ testParams:
     operatorVersion: 2.9.0
     stackVersion: 8.9.0
   - name: upcoming
-    operatorVersion: 2.10.0-SNAPSHOT
-    stackVersion: 8.10.0-SNAPSHOT
+    operatorVersion: 2.11.0-SNAPSHOT
+    stackVersion: 8.11.0-SNAPSHOT
```

Comment:
If we want to also update upgrade-test-harness in this PR, there are more files to update, see for example: https://github.com/elastic/cloud-on-k8s/pull/6865.

Reply:
I've added the missing test-harness data.